### DISCUSSION TITLE
Content provider service is not healthy in knowledge-mw-service

### DISCUSSION THREAD

**Discussion by Snegha-M** [Sunbird-Knowlg/Community#121] [Bugs]
Hi Team
1. As per the code for content provider service in knowlg-mw it is checking three things, isEkStepHealthy && isLSHealthy && isDbConnected
isEkStepHealthy is linked to contentRepoBaseUrl and given a value of (https://qa.ekstep.in/api) which is not existing so can you guide me on what value should be provided for contentRepoBaseUrl Also do we require to set up any other service?
2. While running the code I'm getting an error as 'Error fetching default channel' where it is pointing to the learning service and '/v1/org/search' endpoint in the sunbird content provider library but can't find this endpoint in the learning service.

Please provide a solution and help me to sort out these issues.

**Comment by Snegha-M** [comment]
**Can anyone help me to sort out this issue.**

**Reply by Krishnaj20** [reply]
@maheshkumargangula @pallakartheekreddy - fya please

**Comment by pallakartheekreddy** [comment]
@Snegha-M Can you please share more details
1.  Which version of knowledge-mw-service you are using?
2.  are you doing a local installation or server installation?

**Comment by Snegha-M** [comment]
@pallakartheekreddy 
1. I'm using release-5.2.0
2. Yes I'm doing local installation

**Comment by Snegha-M** [comment]
@pallakartheekreddy 
1. I'm using release-5.2.0
2. Yes I'm doing local installation
3. What are the mandatory fields that need to be changed in environment variables to connect Knowledge-mw-service with the knowledge platform and what are the services that should be running?
4. To make learning service health status true, changed the[ healthCheckService.js file ](https://github.com/Sunbird-Knowlg/knowledge-mw-service/blob/release-5.3.0/src/service/healthCheckService.js) in line 104 removed res.result.response.

Please find the logs below we got from Knowledge-mw-service by checking the URL http://localhost:5000/health,
{
  SHELL: '/bin/bash',
  SESSION_MANAGER: 'local/stl-13123:@/tmp/.ICE-unix/2366,unix/stl-13123:/tmp/.ICE-unix/2366',
  QT_ACCESSIBILITY: '1',
  SNAP_REVISION: '427',
  XDG_CONFIG_DIRS: '/etc/xdg/xdg-ubuntu:/etc/xdg',
  NVM_INC: '/home/snegha/.nvm/versions/node/v16.17.0/include/node',
  XDG_MENU_PREFIX: 'gnome-',
  GNOME_DESKTOP_SESSION_ID: 'this-is-deprecated',
  SNAP_REAL_HOME: '/home/snegha',
  TERMINAL_EMULATOR: 'JetBrains-JediTerm',
  SNAP_USER_COMMON: '/home/snegha/snap/intellij-idea-community/common',
  LANGUAGE: 'en_IN:en',
  PYSPARK_DRIVER_PYTHON: '/usr/bin/python3.8',
  JAVA_HOME: '/usr/lib/jvm/java-11-openjdk-amd64',
  GNOME_SHELL_SESSION_MODE: 'ubuntu',
  SSH_AUTH_SOCK: '/run/user/1000/keyring/ssh',
  TERM_SESSION_ID: '295204a4-11bd-4dd1-8ba1-7443f2dcc216',
  SNAP_INSTANCE_KEY: '',
  XMODIFIERS: '@im=ibus',
  DESKTOP_SESSION: 'ubuntu',
  SSH_AGENT_PID: '2266',
  BAMF_DESKTOP_FILE_HINT: '/var/lib/snapd/desktop/applications/intellij-idea-community_intellij-idea-community.desktop',
  GTK_MODULES: 'Gail:atk-bridge',
  DBUS_STARTER_BUS_TYPE: 'session',
  PWD: '/home/snegha/repo/knowledge-mw-service/src',
  XDG_SESSION_DESKTOP: 'ubuntu',
  LOGNAME: 'snegha',
  XDG_SESSION_TYPE: 'x11',
  GPG_AGENT_INFO: '/run/user/1000/gnupg/S.gpg-agent:0:1',
  XAUTHORITY: '/run/user/1000/gdm/Xauthority',
  DESKTOP_STARTUP_ID: 'org.gnome.Nautilus-2889-stl-13123-env-0_TIME56319',
  SNAP_CONTEXT: 'GEoKmT-piHTdE60ZU6Qk5o2ukD72vsP-BVc84Q3HYpAn2gjMLsEM',
  WINDOWPATH: '2',
  HOME: '/home/snegha',
  USERNAME: 'snegha',
  IM_CONFIG_PHASE: '1',
  LANG: 'en_IN',
  LS_COLORS: 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:',
  XDG_CURRENT_DESKTOP: 'ubuntu:GNOME',
  SNAP_ARCH: 'amd64',
  SNAP_INSTANCE_NAME: 'intellij-idea-community',
  SNAP_USER_DATA: '/home/snegha/snap/intellij-idea-community/427',
  INVOCATION_ID: '814f6aceb5b34b23aee66c02907041bc',
  MANAGERPID: '2028',
  SNAP_REEXEC: '',
  NVM_DIR: '/home/snegha/.nvm',
  PYSPARK_PYTHON: '/usr/bin/python3.8',
  LESSCLOSE: '/usr/bin/lesspipe %s %s',
  XDG_SESSION_CLASS: 'user',
  PYTHONPATH: ':/usr/bin/python3.8',
  TERM: 'xterm-256color',
  SCALA_HOME: '/home/snegha/software/scala-2.12.10',
  LESSOPEN: '| /usr/bin/lesspipe %s',
  USER: 'snegha',
  SNAP: '/snap/intellij-idea-community/427',
  SNAP_COMMON: '/var/snap/intellij-idea-community/common',
  SNAP_VERSION: '2023.1',
  DISPLAY: ':0',
  SHLVL: '1',
  NVM_CD_FLAGS: '',
  SNAP_LIBRARY_PATH: '/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void',
  SNAP_COOKIE: 'GEoKmT-piHTdE60ZU6Qk5o2ukD72vsP-BVc84Q3HYpAn2gjMLsEM',
  SPARK_HOME: '/home/snegha/spark',
  QT_IM_MODULE: 'ibus',
  DBUS_STARTER_ADDRESS: 'unix:path=/run/user/1000/bus,guid=2c4bc0c43e71f13722b704d56437912d',
  SNAP_DATA: '/var/snap/intellij-idea-community/427',
  XDG_RUNTIME_DIR: '/run/user/1000',
  SNAP_NAME: 'intellij-idea-community',
  JOURNAL_STREAM: '8:48841',
  XDG_DATA_DIRS: '/usr/share/ubuntu:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop',
  PATH: '/home/snegha/.nvm/versions/node/v16.17.0/bin:/home/snegha/software/scala-2.12.10/bin:/usr/lib/jvm/java-11-openjdk-amd64/bin:/home/snegha/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/snegha/.local/share/coursier/bin:/home/snegha/spark/bin:/usr/lib/nodejs',
  sunbird_dbs_path: '/home/snegha/sunbird-dbs',
  GDMSESSION: 'ubuntu',
  DBUS_SESSION_BUS_ADDRESS: 'unix:path=/run/user/1000/bus,guid=2c4bc0c43e71f13722b704d56437912d',
  NVM_BIN: '/home/snegha/.nvm/versions/node/v16.17.0/bin',
  GIO_LAUNCHED_DESKTOP_FILE_PID: '2918',
  GIO_LAUNCHED_DESKTOP_FILE: '/var/lib/snapd/desktop/applications/intellij-idea-community_intellij-idea-community.desktop',
  _: '/home/snegha/.nvm/versions/node/v16.17.0/bin/node',
  sunbird_environment: 'dev',
  sunbird_instance: 'sunbird',
  contentServiceBaseUrl: 'http://localhost:9000',
  sunbird_learner_service_local_base_url: 'http://localhost:8080',
  sunbird_default_channel: 'tibil'
}
learnerServiceLocalBaseUrl http://localhost:8080
url http://localhost:8080/learning-service/health
[2023-04-13T11:01:19.077] [INFO] api - {"msg":"keyCloakConfig","keyCloakConfig":{"authServerUrl":"https://staging.open-sunbird.org/auth","realm":"sunbird","clientId":"portal","public":true}}
Telemetry is initialized.
error: null
res: "<!doctype html><html lang=\"en\"><head><title>HTTP Status 404 – Not Found</title><style type=\"text/css\">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 404 – Not Found</h1><hr class=\"line\" /><p><b>Type</b> Status Report</p><p><b>Description</b> The origin server did not find a current representation for the target resource or is not willing to disclose that one exists.</p><hr class=\"line\" /><h3>Apache Tomcat/9.0.72</h3></body></html>"
[2023-04-13T11:01:19.478] [ERROR] api - {"msg":"Error fetching default channel","err":null}
[2023-04-13T11:01:19.479] [INFO] api - {"msg":"DEFAULT_CHANNEL tibil"}
[2023-04-13T11:01:19.479] [INFO] api - {"msg":"keyCloakConfig","keyCloakConfig":{"authServerUrl":"https://staging.open-sunbird.org/auth","realm":"sunbird","clientId":"portal","public":true}}
error while reading publicKey directory: /home/snegha/repo/knowledge-mw-service/src/keys/ [Error: ENOENT: no such file or directory, scandir '/home/snegha/repo/knowledge-mw-service/src/keys/'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: '/home/snegha/repo/knowledge-mw-service/src/keys/'
}
Defaulting to Keycloak validation
[2023-04-13T11:01:19.505] [INFO] api - {"msg":"server running at PORT 5000"}
Kafka Producer is connected and ready.
[2023-04-13T11:01:19.643] [INFO] api - {"msg":"Kafka Producer is connected and ready."}
url http://localhost:8080/learning-service/health
[2023-04-13T11:01:34.105] [INFO] api - {"id":"73e9bad0-d9bc-11ed-9e43-29ade987f021","msg":"CASSANDRA_DB_HEALTH_STATUS is True"}
null
{"id":"learning-service.health","ver":"1.0","ts":"2023-04-13T11:01:34Z+05:30","params":{"resmsgid":"e67f4d5d-9a70-44f2-9f77-178807f069ee","msgid":null,"err":null,"status":"successful","errmsg":null},"responseCode":"OK","result":{"checks":[{"healthy":true,"name":"domain graph"},{"healthy":true,"name":"redis cache"},{"healthy":true,"name":"cassandra db"}],"healthy":true},"statusCode":200}
[2023-04-13T11:01:34.216] [INFO] api - {"id":"73e9bad0-d9bc-11ed-9e43-29ade987f021","msg":"LEARNER_SERVICE_HEALTH_STATUS is True"}
[2023-04-13T11:03:44.940] [ERROR] api - {"msg":"Got error in send Request"}
[2023-04-13T11:03:44.943] [ERROR] api - {"id":"73e9bad0-d9bc-11ed-9e43-29ade987f021","msg":"EKSTEP_HEALTH_STATUS is False","err":true}
[2023-04-13T11:03:44.955] [ERROR] api - {"id":"73e9bad0-d9bc-11ed-9e43-29ade987f021","msg":"Content Service is not healthy","additionalInfo":{"healthStatus":[{"name":"cassandra.db","healthy":true,"err":"","errmsg":""},{"name":"learnerservice.api","healthy":true,"err":"","errmsg":""},{"name":"ekstep.api","healthy":false,"err":"CONTENT_PROVIDER_HEALTH_FAILED","errmsg":"Content provider service is not healthy"}]}}

**Reply by HarishGangula** [reply]
@Snegha-M  From the Above logs **CONTENT_PROVIDER_HEALTH_FAILED**  so check why it is failing and check the config whether it is pointing to the correct endpoint.

**Comment by Snegha-M** [comment]
@HarishGangula 
The content provider health is failing due to the incorrect value of contentRepoBaseUrl in the app.js file
Refer to the below screenshot,
![Screenshot from 2023-04-17 12-26-54](https://user-images.githubusercontent.com/99195190/232408276-73cdac38-6091-4afc-9a4e-086251b6a752.png)
So we want to understand what is contentRepoBaseUrl means and do we need to set up any service for that.

**Reply by pallakartheekreddy** [reply]
@Snegha-M Please pass content-service url for `contentRepoBaseUrl` in app.js

**Comment by Snegha-M** [comment]
@pallakartheekreddy 
Ok thanks..Will get back to you if any issues are there.

**Comment by Snegha-M** [comment]
@pallakartheekreddy 
Now we were able to connect to the content provider but, While running the code I'm getting an error as 'Error fetching default channel' where it is pointing to the learning service and '/v1/org/search' endpoint in the sunbird content provider library but can't find this endpoint in the learning service.
![Screenshot from 2023-04-17 13-09-40](https://user-images.githubusercontent.com/99195190/232418843-c8198bf8-d688-40ec-81a2-c40e105d7f51.png)

**Reply by pallakartheekreddy** [reply]
@Snegha-M it should point to learner service (USER & ORG SERVICE) not learning service. Please try to install USER & ORG SERVICE as as well and point to that service. Please refer below link for more details.
https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service

**Reply by Snegha-M** [reply]
@pallakartheekreddy 
Thanks.Will get back to you if any issues are there.
